### ISSUE TITLE
ws_client.get_wssettings_list KeyError: 'enableNotebookGitVersioning'

### ISSUE THREAD

**Issue by v1adev** [databricks-industry-solutions/security-analysis-tool#151]
The `bootstrap('workspacesettings'+ '_' + workspace_id, ws_client.get_wssettings_list)` call started failing with the following error last week (in Azure Databricks). Could possibly be related to the retirement of the legacy Git version control? https://learn.microsoft.com/en-us/azure/databricks/archive/repos/git-version-control-legacy

```
2024-10-07 12:07:50,562 - _profiler_ - ERROR - Exception encountered
Traceback (most recent call last):
  File "/root/.ipykernel/1714/command-339720792581471-1775977341", line 15, in bootstrap
    lst = func(**kwargs)
          ^^^^^^^^^^^^^^
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.11/site-packages/clientpkgs/ws_settings_client.py", line 65, in get_wssettings_list
    valins['value']=None if valn[keyn['name']] is None else valn[keyn['name']]
                            ~~~~^^^^^^^^^^^^^^
KeyError: 'enableNotebookGitVersioning'
```

**Comment by arunpamulapati** [comment]
Yes this will be disabled as our product removed the the API call in addition to this now is a legacy concept.

**Comment by arunpamulapati** [comment]
Yes, we will need to deep clean it, please stay tuned it will take a bit of time.

**Comment by arunpamulapati** [comment]
We managed to reproduce this issue and it only happens in the newly created workspaces. We missed this in the regression testing as we are using existing workspaces. We are working on a hot-fix.

**Comment by arunpamulapati** [comment]
This now resolved in release/0.3.3 branch.
